src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsJsonEncoding.cs

```diff
@@ -165,11 +152,28 @@ void WriteLaunchProfile(ILaunchProfile profile)
                 foreach ((string key, object value) in profile.EnumerateOtherSettings())
                 {
                     writer.WritePropertyName(key);
-                    writer.WriteValue(value);
+                    WriteObject(value);
```

Comment:
This is the other fix. `writer.WriteValue(object)` cannot handle complex types (such as `Dictionary<,>`). This change extracts the code used to write extra settings to a local function in order to re-use it here.